diff --git a/extra/Configs/Config.in b/extra/Configs/Config.in
index 6b0985e..b02054c 100644
--- a/extra/Configs/Config.in
+++ b/extra/Configs/Config.in
@@ -1442,6 +1442,7 @@ config UCLIBC_HAS_LOCALE
 	bool "Locale Support"
 	select UCLIBC_HAS_WCHAR
 	select UCLIBC_HAS_CTYPE_TABLES
+	select UCLIBC_HAS_LIBICONV
 	help
 	  uClibc now has full ANSI/ISO C99 locale support (except for
 	  wcsftime() and collating items in regex).  Be aware that enabling
