diff --git a/extra/Configs/Config.in b/extra/Configs/Config.in
index b02054c..f312e7a 100644
--- a/extra/Configs/Config.in
+++ b/extra/Configs/Config.in
@@ -1429,6 +1429,7 @@ config UCLIBC_HAS_WCHAR
 
 config UCLIBC_HAS_LIBICONV
 	bool "Iconv Support"
+	select UCLIBC_HAS_WCHAR
 	help
 	  Add tiny iconv support for charset conversion from and to UTF-8.
 
@@ -1821,9 +1822,10 @@ config UCLIBC_HAS_GNU_GETSUBOPT
 	  Most people will answer Y.
 
 config UCLIBC_HAS_ARGP
-	bool "Support argp (as standalone shared object)"
+	bool "Support argp"
 	select UCLIBC_HAS_GETOPT_LONG
 	select UCLIBC_HAS_GNU_GETOPT
+	select UCLIBC_HAS_WCHAR
 	default n
 	help
 	  Argp is an interface for parsing unix-style argument vectors. Unlike
