--- a/libc/sysdeps/linux/or1k/sys/ucontext.h
+++ b/libc/sysdeps/linux/or1k/sys/ucontext.h
@@ -20,6 +20,14 @@
 
 #include <features.h>
 #include <signal.h>
-#include <asm/ucontext.h>
+#include <bits/sigcontext.h>
+
+typedef struct ucontext {
+	unsigned long	  uc_flags;
+	struct ucontext  *uc_link;
+	stack_t		  uc_stack;
+	struct sigcontext uc_mcontext;
+	sigset_t	  uc_sigmask;	/* mask last for extensibility */
+} ucontext_t;
 
 #endif /* sys/ucontext.h */
