--- a/libc/sysdeps/linux/common/preadv.c
+++ b/libc/sysdeps/linux/common/preadv.c
@@ -19,8 +19,10 @@
 #include <sys/syscall.h>
 #include <unistd.h>
 
+#ifdef __NR_preadv
 ssize_t
 preadv (int fd, const struct iovec *vector, int count, off_t offset)
 {
   return INLINE_SYSCALL (preadv, 4, fd, vector, count, offset);
 }
+#endif

--- a/libc/sysdeps/linux/common/pwritev.c
+++ b/libc/sysdeps/linux/common/pwritev.c
@@ -19,8 +19,10 @@
 #include <sys/syscall.h>
 #include <unistd.h>
 
+#ifdef __NR_pwritev
 ssize_t
 pwritev (int fd, const struct iovec *vector, int count, off_t offset)
 {
   return INLINE_SYSCALL (pwritev, 4, fd, vector, count, offset);
 }
+#endif
